### ISSUE TITLE
Helm Chart - ServiceAccount name ignored

### ISSUE THREAD

**Issue by cvockrodt** [oauth2-proxy/manifests#31]
### Problem
The [serviceaccount template](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/templates/serviceaccount.yaml) ignores the value from `serviceaccount.name` and uses the fullname template:

```yaml
{{- if or .Values.serviceAccount.enabled -}}
apiVersion: v1
kind: ServiceAccount
metadata: 
  ...
  name: {{ template "oauth2-proxy.fullname" . }}
{{- end -}}
```

### Expected Behavior
If `serviceaccount.name` is set and `serviceaccount.enabled`=`true`, the serviceaccount in use matches the one that's created.

**Comment by pierluigilenoci** [comment]
There is already a PR to fix this issue (#39)

**Comment by pierluigilenoci** [comment]
Fixed by #39
